### PR TITLE
MakinoSCF

### PR DESCRIPTION
**Pull request by Glinland**

### CODE REVIEW

**Comment by Glinland** [comment]
Pull request

File: SCF.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 SCF.py is a module that contains all of the functions
 for the HF SCF Procedure
 """
-
+from pyscf import gto
```
Comment (shots47s) [inline SCF.py:5]: Should not need this import here.

File: SCF.py [hunk 2]
```diff
@@ -153,8 +165,12 @@ def form_density_matrix(mol_, mo_coeffs_):
...
     nelec = mol_.nelec[0]  # Number of occupied orbitals
     num_aos = mol_.nao  # Number of atomic orbitals, dimensions of the mats
-    Duv = np.zeros(mol_.nao, mol_.nao, dtype=np.double)
+    Duv = np.zeros((mol_.nao, mol_.nao), dtype=np.double)
 
+    for mu in range(num_aos):
+        for nu in range(num_aos):
+            for i in range(nelec):
+                Duv[mu, nu] += (mo_coeffs_[mu, i]*mo_coeffs_[nu, i])*2
     """
```
Comment (shots47s) [inline SCF.py:174]: Extraneous comment block, please remove.

File: SCF.py [hunk 3]
```diff
@@ -180,6 +196,7 @@ def calc_total_energy(Fuv_, Huv_, Duv_, Enuc_):
     Returns:
         Etot: the total energy of the molecule
     """
+    Etot = (Duv_*(Huv_+Fuv_)).sum()/2 + Enuc_
 
     """
```
Comment (shots47s) [inline SCF.py:201]: Extraneous comment block, please remove.

File: SCF.py [hunk 4]
```diff
@@ -153,8 +165,12 @@ def form_density_matrix(mol_, mo_coeffs_):
 
     nelec = mol_.nelec[0]  # Number of occupied orbitals
     num_aos = mol_.nao  # Number of atomic orbitals, dimensions of the mats
-    Duv = np.zeros(mol_.nao, mol_.nao, dtype=np.double)
+    Duv = np.zeros((mol_.nao, mol_.nao), dtype=np.double)
 
+    for mu in range(num_aos):
```
Comment (shots47s) [inline SCF.py:170]: Note: could do with two loops and a numpy sum:

```python
Duv[mu, nu] = 2.0*(mo_coeffs_[mu, :nelec]
                               * mo_coeffs_[nu, :nelec]).sum()
```

File: tests/SCF/test_SCF.py [hunk 5]
```diff
@@ -1,19 +1,89 @@
-import pytest
+from pytest import approx
 import SCF
+import pytest
+from pyscf import gto  # PySCF is a quantum chemistry python module
+import numpy as np
+import sys
+sys.path.append("/Users/mkx/Desktop/Courses/Python/HF_SCF_Assignment")
```
Comment (shots47s) [inline tests/SCF/test_SCF.py:7]: Have no idea why this is here.

File: tests/SCF/test_SCF.py [hunk 6]
```diff
@@ -1,19 +1,89 @@
-import pytest
+from pytest import approx
 import SCF
+import pytest
+from pyscf import gto  # PySCF is a quantum chemistry python module
+import numpy as np
+import sys
+sys.path.append("/Users/mkx/Desktop/Courses/Python/HF_SCF_Assignment")
+print(sys.path)
+mol_h2o = gto.M(unit="Bohr",
```
Comment (shots47s) [inline tests/SCF/test_SCF.py:9]: should not do this, this should use the fixture as was shown in class and through the example function. This is already defined in `conftest.py`

File: tests/SCF/test_SCF.py [hunk 7]
```diff
@@ -1,19 +1,89 @@
...
+import sys
+sys.path.append("/Users/mkx/Desktop/Courses/Python/HF_SCF_Assignment")
+print(sys.path)
+mol_h2o = gto.M(unit="Bohr",
+                atom="O 0.000000000000  -0.143225816552   0.000000000000;"
+                + "H 1.638036840407   1.136548822547  -0.000000000000;"
+                + "H -1.638036840407   1.136548822547  -0.000000000000",
+                basis='STO-3g')
+mol_h2o.build()
+Suv = mol_h2o.intor('int1e_ovlp')  # Overlap Integrals
```
Comment (shots47s) [inline tests/SCF/test_SCF.py:15]: These should not be defined here, they should be called in the actual test functions as this pollutes the global space.

File: tests/SCF/test_SCF.py [hunk 8]
```diff
@@ -1,19 +1,89 @@
...
+    assert results[2, 5] == -1.6751501447185013
+    assert results[0, 0] == -32.57739541261037
+
+# @pytest.fixture()
+# def Fuv():
+#     Fuv = SCF.calc_fock_matrix(mol_h2o,Huv,eri,Duv)
+#     return Fuv
+
+
+Fuv = SCF.calc_fock_matrix(mol_h2o, Huv, eri, Duv)
```
Comment (shots47s) [inline tests/SCF/test_SCF.py:65]: This should not be outside of a function.

File: tests/SCF/test_SCF.py [hunk 9]
```diff
@@ -1,19 +1,89 @@
...
+
+
+def test_solve_Roothan_equations():
+
+    mol_e, mol_c = SCF.solve_Roothan_equations(Fuv, Suv)
+
+    assert mol_e[0] == approx(-32.57830291837955)
+
+
+mol_e, mol_c = SCF.solve_Roothan_equations(Fuv, Suv)
```
Comment (shots47s) [inline tests/SCF/test_SCF.py:75]: Really should test the whole array.

**Review by shots47s — CHANGES_REQUESTED** [review]
SCF code looks good, PEP8 compliance good, the testing code is incorrect, as you redefined the fixture and had some variables outside of the function scope.

**Pull request closed without merging**